dhall-lsp-server/src/Dhall/LSP/Handlers.hs

```diff
@@ -358,6 +362,74 @@ executeAnnotateLet request = do
     (J.ApplyWorkspaceEditParams edit)
 
 
+executeFreezeAllImports :: J.ExecuteCommandRequest -> HandlerM ()
+executeFreezeAllImports request = do
+  uri <- getCommandArguments request
+
+  fileIdentifier <- fileIdentifierFromUri uri
+  txt <- readUri uri
+  expr <- case parse txt of
+    Right e -> return e
+    Left _ -> throwE (Warning, "Could not freeze imports; did not parse.")
+
+  let importRanges = getAllImportsWithHashPositions expr
+  edits <- mapM (\(import_, Range (x1, y1) (x2, y2)) -> do
```

Comment:
Instead of:

```haskell
edits <- mapM (\... -> do
    ...
    ) importRanges
```

... you can do:

```haskell
edits <- forM importRanges $ \... -> do
    ...
```

... where `forM` is from `Control.Monad`